### ISSUE TITLE
Warn volunteers when a link will open in a new tab

### ISSUE THREAD

**Issue by goplayoutside3** [zooniverse/front-end-monorepo#5530]
## Package
_Choose from the list:_
- [ ] app-content-pages
- [ ] app-project
- [ ] lib-classifier
- [ ] lib-panoptes-js
- [x] lib-react-components
- [ ] unknown

## Is your feature request related to a problem? Please describe.

Related to markdown work in https://github.com/zooniverse/front-end-monorepo/pull/5352, it was pointed out that volunteers don't have a warning when external links open a new browser tab.

In Zooniverse-flavored `markdownz`, project teams can add `+tab+` to links to force-open them in a new browser tab. This is PFE's behavior, and we are re-aligning markdown in FEM to do the same (#5352).

## Describe the solution you'd like

Volunteers should have a visual indicator, and blind users should also know a link is going to open (or has opened?) a new browser tab.

One solution outlined in https://css-irl.info/styling-external-links-with-attribute-selectors is to style links that contain `http` with an "external link" icon via CSS. However, that might not be a fully accessible solution.

I'm opening this Issue and tagging both @seanmiller26 to consider placement of an "external link" icon, and @eatyourgreens for continued discussion on accessibility of external links in markdown. When design and code solution is decided, we'll use this Issue to implement it in FEM's [Markdownz](https://github.com/zooniverse/front-end-monorepo/tree/master/packages/lib-react-components/src/Markdownz) component.

**Comment by eatyourgreens** [comment]
Here's the WCAG advice on [opening new windows only when necessary](https://www.w3.org/WAI/WCAG21/Techniques/general/G200).

At one point, we modified Snapshot Serengeti so that all links opened in a new tab. Volunteers hated that.

**Comment by eatyourgreens** [comment]
Project links open in the same tab, which is in keeping with the general advice to avoid new tabs unless following a link would interrupt work-in-progress eg. a partially filled-out classification form. 

<img width=400 src="https://github.com/zooniverse/front-end-monorepo/assets/59547/3a255f08-b010-43dd-94ef-3057e8ea0541">

**Comment by goplayoutside3** [comment]
Linking this issue here too: https://github.com/zooniverse/front-end-monorepo/issues/3633 External Links section or "Connect with" links not opening in new tab. I think the solution to that Issue may be to modify lab instructions rather than force-open a new tab for that page section.

**Comment by seanmiller26** [comment]
Looking more into use cases here I think it's important for certain links to open in new tabs. Here are some great examples: 
[Talk example](https://www.zooniverse.org/talk/18/557175?comment=923370)   [Another](https://www.zooniverse.org/talk/17/1724916) [One more](https://www.zooniverse.org/talk/18/1190649?comment=1939056&page=1)

It is a heavily used feature so I agree we must bring it to FEM. For cases like these our standard should be to use the Grommet icon "Share" at the end of the link. 

Links can also appear in banners and metadata fields, so we'll want to make sure we're consistent in our usage. I'm assuming that since this will impact Markdownz globally, this should take care of all areas.
[I also came across this](https://www.zooniverse.org/talk/18/557175?comment=1293930&page=2)

 
As far as the Connect section - I'm in favor of giving the teams the option of opening links in either orientation. Some links are helpful while classifying, while others are further reading which take users away from our site. We may need to add a dropdown or similar to the lab section for this.

**Comment by eatyourgreens** [comment]
@seanmiller26 links within a page should never open in a new tab. They should just scroll to the appropriate section. See the links at the top of the Penguin Watch FAQ for examples. 
https://fe-project-branch.preview.zooniverse.org/projects/penguintom79/penguin-watch/about/faq

**Comment by eatyourgreens** [comment]
Per the WCAG advice I linked above, it is fine to open a link like the Measuring The Anzacs Abbrevations in a new tab if opening in the same tab would interrupt work-in-progress.

**Comment by eatyourgreens** [comment]
Here's a demo of the problem with links to project pages at the moment. The second link to the project Team page will open in a new tab, because Remark opens all absolute URLs in new tabs.
https://frontend.preview.zooniverse.org/projects/eatyourgreens/-project-testing-ground/about/research?env=staging

**Comment by goplayoutside3** [comment]
I think it might be helpful to narrow this conversation to what links on project pages we DO want to open in a new tab - and then implement a warning icon or similar.

So far we do want
1. links in markdown with `+tab+` to open in a new tab (field guide, tutorial, about pages, etc)
2. meta data links to open in a new tab

Any others?

**Comment by seanmiller26** [comment]
Nothing else I can think of at this time.

Some volunteers open Talk in another tab but I definitely don't think that should be the default. They can continue to command-click or right click in these cases.

**Comment by eatyourgreens** [comment]
[How to make external links accessible](https://blog.pope.tech/2024/01/02/how-to-make-external-links-accessible/).
